### PR TITLE
[43] TSV creation fails due to array fields

### PR DESCRIPTION
**Pull request by lrivera-oicr**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -79,6 +79,7 @@ spec:
             when {
                 anyOf {
                     branch 'develop'
+                    branch 'arrays'
```
Comment (caravinci) [inline Jenkinsfile:82]: We won't need these once the branch is merged & deleted.
```suggestion
```

File: Jenkinsfile [hunk 2]
```diff
@@ -100,6 +101,7 @@ spec:
...
```
Comment (caravinci) [inline Jenkinsfile:104]:
```suggestion
```

File: src/main/java/org/cancogenvirusseq/singularity/components/model/AnalysisDocument.java [hunk 3]
```diff
@@ -41,11 +45,12 @@ public class AnalysisDocument {
...
   public static final String LAST_UPDATED_AT_FIELD = "analysis.updated_at";
 
-  @NonNull private String objectId;
-  @NonNull private String studyId;
+  @NonNull private JsonNode objectId;
+  @NonNull private JsonNode studyId;
   @NonNull private Analysis analysis;
   @NonNull private List<Donor> donors;
 
+
```
Comment (caravinci) [inline src/main/java/org/cancogenvirusseq/singularity/components/model/AnalysisDocument.java:53]:
```suggestion
```

File: src/main/java/org/cancogenvirusseq/singularity/components/utils/TsvUtils.java [hunk 4]
```diff
@@ -19,15 +19,22 @@
...
 import java.util.List;
 import java.util.Optional;
 import java.util.stream.Collectors;
 import javax.annotation.Nullable;
+
+import com.fasterxml.jackson.databind.JsonNode;
 import lombok.Getter;
 import org.cancogenvirusseq.singularity.components.model.AnalysisDocument;
 
+
```
Comment (caravinci) [inline src/main/java/org/cancogenvirusseq/singularity/components/utils/TsvUtils.java:33]:
```suggestion
```

**Review by caravinci — CHANGES_REQUESTED** [review]
Some preliminary comments and general style nitpicking.

File: src/main/java/org/cancogenvirusseq/singularity/components/model/AnalysisDocument.java (hunk 3 above)
Comment (UmmulkiramR) [reply src/main/java/org/cancogenvirusseq/singularity/components/model/AnalysisDocument.java:53]: JsonProcessingException in the AnalysisDocument class can be removed from the imports if not being used.

File: src/main/java/org/cancogenvirusseq/singularity/components/utils/TsvUtils.java (hunk 4 above)
Comment (UmmulkiramR) [reply src/main/java/org/cancogenvirusseq/singularity/components/utils/TsvUtils.java:33]: That's good work! 
You can may be change the indentation in TsvUtils to 2 spaces as that is the standard followed.

**Review by UmmulkiramR — CHANGES_REQUESTED** [review]

File: src/main/java/org/cancogenvirusseq/singularity/components/utils/TsvUtils.java (hunk 4 above)
Comment (UmmulkiramR) [reply src/main/java/org/cancogenvirusseq/singularity/components/utils/TsvUtils.java:33]: How about making the LIST_SEPARATOR configurable?

File: Jenkinsfile (hunk 1 above)
Comment (lrivera-oicr) [reply Jenkinsfile]: arrays branch removed from Jenkinsfile

File: src/main/java/org/cancogenvirusseq/singularity/components/utils/TsvUtils.java (hunk 4 above)
Comment (lrivera-oicr) [reply src/main/java/org/cancogenvirusseq/singularity/components/utils/TsvUtils.java]: added indentation and LIST_SEPARATOR is now configurable from application.yaml  utils.tsvListSeparator property

File: src/main/java/org/cancogenvirusseq/singularity/components/model/AnalysisDocument.java [hunk 5]
```diff
@@ -78,98 +81,101 @@ public void setFirstPublishedAt(String firstPublishedAt) {
...
-    private String hostScientificName;
+    private JsonNode hostAge;
+    private JsonNode hostAgeNullReason;
+    private JsonNode hostGender;
+    private JsonNode hostAgeBin;
+    private JsonNode hostDisease;
+    private JsonNode hostAgeUnit;
+    private JsonNode hostScientificName;
+
+
```
Comment (caravinci) [inline src/main/java/org/cancogenvirusseq/singularity/components/model/AnalysisDocument.java:112]:
```suggestion
```

**Review by caravinci — APPROVED** [review]
Ready to "Squash and Merge" 🚢

**Pull request merged**